src/core/lb.cpp

```diff
@@ -1254,10 +1263,13 @@ int lb_lbnode_get_pi_neq(int* ind, double* p_pi) {
 int lb_lbnode_get_boundary(int* ind, int* p_boundary) {
     if (lattice_switch & LATTICE_LB_GPU) {
 #ifdef LB_GPU
-        unsigned int host_flag;
+        int host_flag;
         int single_nodeindex = ind[0] + ind[1]*lbpar_gpu.dim_x + ind[2]*lbpar_gpu.dim_x*lbpar_gpu.dim_y;
         lb_get_boundary_flag_GPU(single_nodeindex, &host_flag);
```

Comment:
host_flag is set by this function. A wrapper function is used since the data has to be read via a cuda Kernelcall.